File: 002/prob2.c
//By considering the terms in the Fibonacci 
//sequence whose values do not exceed four million, 
//find the sum of the even-valued terms.
#include <stdio.h>

int
main() {
	//generate the fibonacci numbers that are 
	//less than 4,000,000
	int j = 0;
	int k = 1;
	int sum = 0;

	while (j < 4000000)
	{
		j += k;
		k += j;

		if (j % 2 == 0 && j < 4000000) sum += j;
		if (k % 2 == 0 && k < 4000000) sum += k;
	}

	printf("%d\n", sum);

	return 0;
}

File: 002/prob2.py
#Project Euler Prolem 2
#By considering the terms in the Fibonacci sequence whose values do 
#not exceed four million, find the sum of the even-valued terms.

def getFibonacciSequence():
	fib = [1, 2]
	
	while fib[len(fib)-1]+fib[len(fib)-2] < 4000000:
		fib.append(fib[len(fib)-1]+fib[len(fib)-2])
	
	return fib

def isEven(checkMe):
	returnValue = False
	
	if checkMe % 2 == 0:
		returnValue = True
	
	return returnValue

fib = getFibonacciSequence()
sum = 0
i = 0

while i < len(fib):
	if isEven(fib[i]):
		sum += fib[i]
	
	i += 1

print sum
